sdks/java/harness/src/test/java/org/apache/beam/fn/harness/state/StateFetchingIteratorsTest.java

```diff
@@ -268,7 +267,7 @@ public void testManyValues() throws Exception {
           assertEquals(Math.min(i + 2, expected.length), callCount.get());
         }
         assertTrue(valuesIter2.hasNext());
-        results.add(valuesIter2.next());
+        valuesIter2.next();
       }
       assertFalse(valuesIter2.hasNext());
       assertTrue(valuesIter2.isReady());
```

Comment:
I asked about this over in https://github.com/apache/beam/pull/15475/files/0ebe6567a2057cdb642833c5c9d28853ec5ecdd6#r762215539

It seems the intention was to validate `results` like this:
```suggestion
      assertTrue(valuesIter2.isReady());
      assertArrayEquals(expected, Iterables.toArray(results, Object.class));
```

Reply:
Can you try adding that? If it makes the test fail let's file a jira to follow up on this.

Reply:
The test is passing with these changes, thanks.